insights/client/utilities.py

```diff
@@ -52,27 +52,51 @@ def get_time():
     return datetime.datetime.isoformat(datetime.datetime.now())
 
 
+def write_registered_file():
+    delete_unregistered_file()
+    for f in constants.registered_files:
+        if os.path.lexists(f):
+            if os.path.islink(f):
+                # kill symlinks and regenerate
+                os.remove(f)
+                write_to_disk(f)
+        else:
+            write_to_disk(f)
```

Comment:
nit: seems like the logic could be a bit simpler:

```
if os.path.islink(f):
    os.remove(f)
write_to_disk(f)
```

Reply:
Nevermind, I now see that your code doesn't write to disk when the file exists and is not a symlink.  There's probably no harm in writing the file again if it already exists, but either way is fine.

Reply:
The main reason is because we write the current time as time of registration to the file and regenerating it every time would make that data meaningless